File: BACK-END/KuentecoApp/src/main/java/org/kuenteco/backend/dto/logic/category/NewCategoryDTO.java
package org.kuenteco.backend.dto.logic.category;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.kuenteco.backend.entity.extra.DescriptionCategory;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class NewCategoryDTO {
    @Positive(message = "El ID del presupuesto debe ser un número positivo")
    private Integer budgetId;

    @NotNull private String name;

    @Valid
    @NotNull(message = "La descripción de la categoría es requerida")
    private DescriptionCategory description;
}

File: FRONT-END/lib/dto/app/category/new_category_dto.dart
import '../extra/description_category_extra.dart';

class NewCategoryDTO {
  final int? budgetId;
  final String name;
  final DescriptionCategory description;

  NewCategoryDTO({
    this.budgetId,
    required this.name,
    required this.description,
  });

  factory NewCategoryDTO.fromJson(Map<String, dynamic> json) {
    return NewCategoryDTO(
      budgetId: json['budgetId'],
      name: json['name'],
      description: DescriptionCategory.fromJson(json['description']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'budgetId': budgetId,
      'name': name,
      'description': description.toJson(),
    };
  }
}
